subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ApplicationPluginConfigurationIntegrationTest.groovy

```diff
@@ -109,10 +106,9 @@ class ApplicationPluginConfigurationIntegrationTest extends AbstractIntegrationS
         out.toString() == TextUtil.toPlatformLineSeparators("Module: $expectedModule\n")
 
         where:
-        configClass                   | configModule                  | expectedModule | deprecation
-        "mainClassName = 'test.Main'" | ''                            | 'null'         | true
-        "mainClass.set('test.Main')"  | ''                            | 'null'         | false
-        "mainClass.set('test.Main')"  | "mainModule.set('test.main')" | 'test.main'    | false
-        ''                            | "mainModule.set('test.main')" | 'test.main'    | false
+        configClass                  | configModule                  | expectedModule
```

Comment:
Now that we don't have the `mainClassName =` part here, do you think it's worth refactoring these to use booleans or something instead of pasting strings into the build file?

Reply:
I think that breaking up the setup for the build script into multiple writes with branching logic controlling them would be less clear than the variable substitutions done here.